README.md

```diff
@@ -211,8 +213,25 @@ $records = $porter->import(
         ->setCacheAdvice(CacheAdvice:SHOULD_CACHE())
 );
 ```
+### Cache key
+
+The cache key can optionally be generated by an implementation of `CacheKeyGeneratorInterface` passed into `CachingConnector::fetch`. This implementation should provide one method `generateCacheKey` which returns a [PSR-6][PSR-6]-compatible cache key.
+
+The default cache key is comprised of the source and options parameters passed to `Connector::fetch`. Options are sorted before the cache key is created so the order of options are insignificant.
+
+#### Example
+
+```php
+class SingleCacheKeyGenerator implements CacheKeyGeneratorInterface
+{
+    public function generateCacheKey($source, EncapsulatedOptions $options) {
+        return 'KEY';
+    }
+}
 
+```
 Architecture
+
```

Comment:
I think this blank line was supposed to go above this heading rather than beneath it?